### ISSUE TITLE
BarChart enable possibility to label axes.

### ISSUE THREAD

**Issue by SamTV12345** [primefaces/primefaces#8705]
**Description**
Add labels for the x and y-axis of the charts. Concerning the showcase http://primefaces.org/showcase/ui/chartjs/bar/bar.xhtml? I don't know anything about the unit of the displayed numbers.

**Comment by SamTV12345** [comment]
Found out that I should use the method setExtender. It would be helpful to mention that in the showcase.

**Comment by melloware** [comment]
I think it's mentioned in the PrimeFaces docs for charts right?  There is a whole section on using the Extender?

**Comment by melloware** [comment]
Also can you post your code that you used in the extender to show the labels?  Did you need to add any special plugins?

**Comment by SamTV12345** [comment]
In  the chartJS docs they mention to use https://www.chartjs.org/docs/latest/configuration/tooltip.html. Unfortunately this only inserts the dataset's label into the tooltip. In my usecase I used a barChart with percentage values.  It would be great if you could insert an extender sample into the showcase as labelling axis and modifying the tooltips is quite important. 


```javascript

            function customizeTotalProgressBar() {
                let options = {
                    options: {
                        scales: {
                            y: {
                                title: {
                                    text:"Quiz",  // Modify y-axis title 
                                    display: true // Important otherwise it is not showing up.
                                }
                            },
                            x:{
                                title: {
                                    text: "Fortschritt in Prozent", // Modify title 
                                    display: true // Important otherwise it is not showing up.
                                },
                                suggestedMin:0, // minimum value that will always be relied on except that there are values smaller 0
                                suggestedMax: 100 //maximum value that will always be relied on except that there are values greater 100
                            }
                        },
                    plugins: {
                            tooltip: {
                                callbacks: {
                                    label: function (context) {
                                       return context.dataset.data[context.dataIndex] +"%" //data is an array of values of the chart. To get the current index, use context.dataIndex
                                    }
                                }
                            }
                        }
                    }
                }
                //merge all options into the main chart options
                $.extend(true, this.cfg.config, options)
            }
```

**Comment by melloware** [comment]
This part is all fixed in 12.0.0 you can set these in the Java model object.

```js
y: {
                                title: {
                                    text:"Quiz",  // Modify y-axis title 
                                    display: true // Important otherwise it is not showing up.
                                }
                            },
                            x:{
                                title: {
                                    text: "Fortschritt in Prozent", // Modify title 
                                    display: true // Important otherwise it is not showing up.
                                },
                                suggestedMin:0, // minimum value that will always be relied on except that there are values smaller 0
                                suggestedMax: 100 //maximum value that will always be relied on except that there are values greater 100
                            }
```

The tooltip callback function you will always need the extender for.  since it requires Javascript code.

We document the extender here: https://primefaces.github.io/primefaces/11_0_0/#/components/charts?id=extender

But showing examples of every sing ChartJS combination is not quite possible.

Another plugin you might be interested in is chartjs-data-labels it diplays the values right on the charts themselves without needing a tooltip: https://chartjs-plugin-datalabels.netlify.app/guide/#table-of-contents

See this link on how to enable DataLabels for ChartsJS: https://stackoverflow.com/a/55765721/502366

**Comment by SamTV12345** [comment]
Thank you for your help. Great that this is fixed in 12.0. It is not very comfortable to do I18n in two systems. I will take a look at this plugin. I need to test if the length of the strings will surpass the individual tiles, especially in the PieCharts.
